lib/all.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var R = require('ramda');
+
+/**
+ * Output a block (or its inverse) based on whether or not all of the supplied
+ * arguments are truthy.
+ *
+ * @since v0.11.0
+ * @param {...*} values One or more values to test against.
+ * @param {Object} options
+ * @return {String|Boolean}
+ * @example
+ * var a = true;
+ * var b = 1;
+ * var c = false;
+ *
+ * {{#all a b}}✔︎{{else}}✘{{/all}} //=> ✔︎
+ * {{#all b c}}✔︎{{else}}✘{{/all}} //=> ✘
+ * 
+ * {{#if (all a b)}}
+ *   Also works inline!
+ * {{/if}}
+ */
+
+function all() {
+  var values = R.dropLast(1, arguments);
+  var options = R.last(arguments);
+  var result = true;
+  var i;
+
+  for (i = 0; i < values.length; i++) {
```

Comment:
@tylersticka Can this be simplified to use [`Array.prototype.every`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)? 🤔 

The [`Array.prototype.every` browser support](http://kangax.github.io/compat-table/es5/#test-Array_methods_Array.prototype.every_a_href=_https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every_title=_MDN_documentation_img_src=_../mdn.png_alt=_MDN_(Mozilla_Development_Network)_logo_width=_15_height=_13_/_/a_nbsp;) seems solid. 🔍  

Instead of a `for` loop, you might have something like:
```js
result = values.every(value => value);
```
According to [the "truthy" MDN docs](https://developer.mozilla.org/en-US/docs/Glossary/Truthy):
> In JavaScript, a truthy value is a value that is considered true when encountered in a Boolean context.

In combination with the [`Array.prototype.every` MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every): 
> It returns a Boolean value.

This to me, unless I'm mistaken, means you can return the `value` directly in the `Array.prototype.every` callback because the truthy values are being evaluated within a Boolean context. 🙂

Reply:
@gerardo-rodriguez I'm torn because, currently, this whole repo seems _very_ old school in its structure. Whenever I've reached a fork in the road between something new or something I'm familiar with, its current code style has made me feel like the more familiar route is the way to go.

But that browser table does certainly support your idea.

Reply:
FWIW the [EcmaScript spec that included `Array.prototype.every`](https://www.ecma-international.org/ecma-262/5.1/#sec-15.4.4.16) was written in 2011 😄. I definitely get wanting to stick to what's familiar, but I wouldn't call `Array.prototype.every` _new_

Reply:
_Ouch_, @calebeby! We can't all be wunderkinds, you know. 😉 

![](http://lol.tylersticka.com/fellowkids.gif)